### PR TITLE
transactions for follow-up requests

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR shows transactions for follow-up requests.

### CODE REVIEW

File: skyportal/handlers/api/source.py [hunk 1]
```diff
@@ -241,6 +242,7 @@ async def get_source(
                     joinedload(FollowupRequest.allocation).joinedload(Allocation.group),
                     joinedload(FollowupRequest.requester),
                     joinedload(FollowupRequest.watchers),
+                    joinedload(FollowupRequest.transactions),
```
Comment (Theodlz) [inline skyportal/handlers/api/source.py:245]: You can specify the column too in the joinedload with `joinonly`. that way you don't query all the stuff you don't need (so you can get only the content that you use after).

File: skyportal/handlers/api/source.py [hunk 2]
```diff
@@ -249,6 +251,22 @@ async def get_source(
         .unique()
         .all()
     )
+
+    followup_requests_data = []
+    for req in followup_requests:
+        req_data = req.to_dict()
```
Comment (Theodlz) [inline skyportal/handlers/api/source.py:257]: no need for the to_dict() here, that serializes all the columns you don't use too.

**Review by Theodlz — CHANGES_REQUESTED** [review]
small comments to avoid loading data we don't need.

File: skyportal/handlers/api/source.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/source.py:260]: These requests were already in the source, so I think we want this.

File: skyportal/handlers/api/source.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/source.py]: added.

File: skyportal/handlers/api/source.py (hunk 2 above)
Comment (Theodlz) [reply skyportal/handlers/api/source.py:260]: Oh right apologies I thought the to_dict ran on on the transaction, my bad!

**Review by Theodlz — APPROVED** [review]
LGTM. One thing I am worried about is that some instruments might leak secrets from the allocation in their API response. I vaguely remember this happening for one or two instruments already, but I guess we'll see :). If this is just for debug and not everyone needs one, we should consider only querying and returning the transactions if the user is an admin, that might make more sense.

**Comment by mcoughlin** [comment]
@Theodlz great idea

**Pull request merged**